src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/Models/ReindexJobRecord.cs

```diff
@@ -119,10 +121,10 @@ protected ReindexJobRecord()
         public ConcurrentDictionary<ReindexJobQueryStatus, byte> QueryList { get; private set; } = new ConcurrentDictionary<ReindexJobQueryStatus, byte>();
 
         [JsonProperty(JobRecordProperties.ResourceCounts)]
-        public ConcurrentDictionary<string, int> ResourceCounts { get; private set; } = new ConcurrentDictionary<string, int>();
+        public ConcurrentDictionary<string, SearchResultReindex> ResourceCounts { get; private set; } = new ConcurrentDictionary<string, SearchResultReindex>();
```

Comment:
This creates the possibility that previous records won't deserialize correctly, did we test this scenario?

Reply:
Great catch on this. I have now added a converter to address this situation as well as a test that demonstrates that an older version of the ReindexJobRecord json can still be deserialized correctly.